File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/payment/application/query/GetRefundReasonInsightsService.java
package com.example.accountservice.payment.application.query;

import java.time.LocalDate;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * Queries refund-reason category counts for ops analytics. Unlike {@link GetRefundsService}, this
 * is deliberately not scoped by requesterId/ownerId — see {@link RefundReasonInsightsQuery}'s
 * Javadoc for why.
 */
@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class GetRefundReasonInsightsService {

    private final RefundReasonInsightsQuery refundReasonInsightsQuery;

    public RefundReasonInsightsResult getInsights(LocalDate fromDate, LocalDate toDate) {
        return refundReasonInsightsQuery.getInsights(fromDate, toDate);
    }
}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/payment/application/query/GetRefundReasonInsightsService.kt
package com.example.accountservice.payment.application.query

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.time.LocalDate

/**
 * Queries refund-reason category counts for ops analytics. Unlike [GetRefundsService], this is
 * deliberately not scoped by requesterId/ownerId — see [RefundReasonInsightsQuery]'s KDoc for why.
 */
@Service
@Transactional(readOnly = true)
class GetRefundReasonInsightsService(
    private val refundReasonInsightsQuery: RefundReasonInsightsQuery,
) {
    fun getInsights(
        fromDate: LocalDate?,
        toDate: LocalDate?,
    ): RefundReasonInsightsResult = refundReasonInsightsQuery.getInsights(fromDate, toDate)
}
